### PR TITLE
[YUNIKORN-2446] Add OCI annotations to public docker images

### PR DESCRIPTION
**Pull request by ryankert01**
### What is this PR for?
source: https://github.com/apache/yunikorn-web/pull/174#issuecomment-1959233281

OCI annotations: https://github.com/opencontainers/image-spec/blob/main/annotations.md

I believe most Yunikorn users are using the docker images into which we push, and so we should consider following a public protocol to set attributions for our public images.

```
org.opencontainers.image.title="yunikorn-scheduler-k8s"
org.opencontainers.image.description="Apache Yunikorn"
org.opencontainers.image.created="${build_date}"
org.opencontainers.image.source="https://github.com/apache/yunikorn-k8shim"
org.opencontainers.image.url="https://hub.docker.com/r/apache/yunikorn"
org.opencontainers.image.revision="${build_revision}"
org.opencontainers.image.license="Apache-2.0"
org.opencontainers.image.documentation="https://yunikorn.apache.org"
```

Reference:
[1] [Annotations and Labels in Container Images](https://adrianmouat.com/posts/annotations-and-labels-in-container-images/)
[2] [OCI annotation spec](https://github.com/opencontainers/image-spec/blob/8797c3fedb77200c102c816ddcd295bb1e19d3e3/annotations.md)

### What type of PR is it?
* [ ] - Bug Fix
* [ ] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2446

#### Make all 3 images
```
make image
```

#### Inspect if oci annotations is added into `config.labels`
```
docker image inspect --format='{{json .Config.Labels}}' apache/yunikorn:scheduler-amd64-latest | jq
```
```
docker image inspect --format='{{json .Config.Labels}}' apache/yunikorn:scheduler-plugin-amd64-latest | jq
```
```
docker image inspect --format='{{json .Config.Labels}}' apache/yunikorn:admission-amd64-latest | jq
```
### Screenshots (if appropriate)

image

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -501,6 +501,14 @@ sched_image: $(OUTPUT)/third-party-licenses.md scheduler docker/scheduler
 	--label "yunikorn-k8shim-revision=${SHIM_SHA}" \
 	--label "BuildTimeStamp=${DATE}" \
 	--label "Version=${VERSION}" \
+	--label org.opencontainers.image.title="yunikorn-scheduler-k8s" \
+	--label org.opencontainers.image.description="Apache Yunikorn Scheduler" \
```
Comment (craigcondit) [inline Makefile:505]: Apache Yunikorn Scheduler -> Apache YuniKorn Scheduler (capitalization)

File: Makefile [hunk 2]
```diff
@@ -570,6 +586,14 @@ adm_image: $(OUTPUT)/third-party-licenses.md admission docker/admission
 	--label "yunikorn-k8shim-revision=${SHIM_SHA}" \
 	--label "BuildTimeStamp=${DATE}" \
 	--label "Version=${VERSION}" \
+	--label org.opencontainers.image.title="yunikorn-admission-controller-k8s" \
+	--label org.opencontainers.image.description="Apache Yunikorn Admission Controller" \
```
Comment (craigcondit) [inline Makefile:590]: Apache Yunikorn Admission Controller -> Apache YuniKorn Admission Controller (capitalization)

File: Makefile [hunk 3]
```diff
@@ -522,6 +530,14 @@ plugin_image: $(OUTPUT)/third-party-licenses.md plugin docker/plugin conf/schedu
 	--label "yunikorn-k8shim-revision=${SHIM_SHA}" \
 	--label "BuildTimeStamp=${DATE}" \
 	--label "Version=${VERSION}" \
+	--label org.opencontainers.image.title="yunikorn-scheduler-plugin-k8s" \
+	--label org.opencontainers.image.description="Apache Yunikorn Plugin" \
```
Comment (craigcondit) [inline Makefile:534]: Apache Yunikorn Plugin -> Apache YuniKorn Scheduler (Plugin)  -- (capitalization / wording)

**Review by craigcondit — CHANGES_REQUESTED** [review]
The idea is sound.

I think we should add `org.opencontainers.image.version` to match our existing `Version` label as well.

We also should update `image.source` and `image.url` to be sourced from environment variables to allow customizing these as part of release builds. We don't want this metadata embedded if usera are generating custom binaries. 

Finally, please open a parallel PR for the yunikorn-web repo so that we get that updated as well.

File: Makefile (hunk 1 above)
Comment (ryankert01) [reply Makefile]: Done.

File: Makefile (hunk 2 above)
Comment (ryankert01) [reply Makefile]: Done.

File: Makefile (hunk 3 above)
Comment (ryankert01) [reply Makefile]: Done.

**Comment by ryankert01** [comment]
Thanks @craigcondit for the review! I also add more variables to increase consistency and reusability.

**Review by craigcondit — COMMENTED** [review]
This looks reasonable to me. @wilfred-s can you take a look as well?

**Review by craigcondit — APPROVED** [review]
Approved (tentatively pending Wilfred's comments)

**Review by pbacsko — APPROVED** [review]
LGTM

**Review by chia7712 — APPROVED** [review]
@ryankert01 thanks for this patch. LGTM

**Pull request closed without merging**